crates/biome_js_analyze/src/lint/nursery/no_vue_setup_props_reactivity_loss.rs

```diff
@@ -0,0 +1,775 @@
+            if seen.insert(range) {
+                deduped.push(f);
+            }
+        }
+        deduped
+    }
+}
+
+fn extract_function_from_expression(expr: &AnyJsExpression) -> Option<AnyJsFunction> {
+    match expr {
+        AnyJsExpression::JsFunctionExpression(func_expr) => {
+            Some(AnyJsFunction::JsFunctionExpression(func_expr.clone()))
+        }
+        AnyJsExpression::JsArrowFunctionExpression(arrow_func) => {
+            Some(AnyJsFunction::JsArrowFunctionExpression(arrow_func.clone()))
+        }
+        _ => None,
+    }
+}
+
+fn extract_setup_from_object_member(
+    member: &biome_js_syntax::AnyJsObjectMember,
+) -> Option<SetupFunction> {
+    use biome_js_syntax::AnyJsObjectMember;
+
+    match member {
+        AnyJsObjectMember::JsMethodObjectMember(method) => {
+            let name = method.name().ok()?;
+            is_member_named_setup(&name).then(|| SetupFunction::Method(method.clone()))
+        }
+        AnyJsObjectMember::JsPropertyObjectMember(property) => {
+            let name = property.name().ok()?;
+            if !is_member_named_setup(&name) {
+                return None;
+            }
+            let value = property.value().ok()?;
+            extract_function_from_expression(&value).map(SetupFunction::Function)
+        }
+        _ => None,
+    }
+}
+
+fn is_member_named_setup(name: &biome_js_syntax::AnyJsObjectMemberName) -> bool {
+    match name {
+        biome_js_syntax::AnyJsObjectMemberName::JsLiteralMemberName(lit) => {
+            lit.value().is_ok_and(|tok| {
+                let text = tok.text_trimmed();
+                if (text.starts_with('"') && text.ends_with('"'))
+                    || (text.starts_with('\'') && text.ends_with('\''))
+                {
+                    let inner = &text[1..text.len() - 1];
+                    inner == "setup"
+                } else {
+                    text == "setup"
+                }
+            })
+        }
+        biome_js_syntax::AnyJsObjectMemberName::JsComputedMemberName(computed) => {
+            computed.expression().is_ok_and(|expr| {
+                let AnyJsExpression::AnyJsLiteralExpression(literal_expr) = expr else {
+                    return false;
+                };
+                let Some(string_literal) = literal_expr.as_js_string_literal_expression() else {
+                    return false;
+                };
+                string_literal.value_token().is_ok_and(|tok| {
+                    let text = tok.text_trimmed();
+                    if text.len() >= 2 {
+                        let inner = &text[1..text.len() - 1];
+                        inner == "setup"
+                    } else {
+                        false
+                    }
+                })
+            })
+        }
+        biome_js_syntax::AnyJsObjectMemberName::JsMetavariable(_) => false,
+    }
+}
+fn get_function_first_parameter(func: &SetupFunction) -> Option<AnyJsBindingPattern> {
```

Comment:
The function returns a binding, not a pattern. You might want to document the function, because the same isn't descriptive enough